fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/ReplacePropertiesServiceImpl.java

```diff
@@ -0,0 +1,217 @@
+import org.fcrepo.kernel.api.models.FedoraResource;
+import org.fcrepo.kernel.api.operations.RdfSourceOperationFactory;
+import org.fcrepo.kernel.api.operations.ResourceOperation;
+import org.fcrepo.kernel.api.services.ReplacePropertiesService;
+import org.fcrepo.persistence.api.PersistentStorageSession;
+import org.fcrepo.persistence.api.PersistentStorageSessionManager;
+import org.fcrepo.persistence.api.exceptions.PersistentStorageException;
+import org.slf4j.Logger;
+
+/**
+ * This class mediates update operations between the kernel and persistent storage layers
+ * @author bseeger
+ */
+
+public class ReplacePropertiesServiceImpl implements ReplacePropertiesService {
+
+    private static final Logger LOGGER = getLogger(ReplacePropertiesServiceImpl.class);
+
+    private static final Node WEBAC_ACCESS_TO_URI = createURI(WEBAC_ACCESS_TO);
+
+    private static final Node WEBAC_ACCESS_TO_CLASS_URI = createURI(WEBAC_ACCESS_TO_CLASS);
+
+    @Inject
+    private RdfSourceOperationFactory factory;
+
+    @Inject
+    private PersistentStorageSessionManager psManager;
+
+    @Override
+    public void replaceProperties(final Transaction tx,
+                           final FedoraResource fedoraResource,
+                           final InputStream requestBodyStream,
+                           final String contentType) throws MalformedRdfException {
+        try {
+            final PersistentStorageSession pSession = this.psManager.getSession(tx.getId());
+
+            final Model inputModel = parseBodyAsModel(requestBodyStream, contentType, fedoraResource);
+
+            ensureValidMemberRelation(inputModel);
+
+            ensureValidACLAuthorization(fedoraResource, inputModel);
+
+            final ResourceOperation updateOp = factory.updateBuilder(fedoraResource.getId())
+                .triples(fromModel(createURI(fedoraResource.getId()), inputModel))
+                .build();
+
+            pSession.persist(updateOp);
+        } catch (final PersistentStorageException ex) {
+            throw new RepositoryRuntimeException(String.format("failed to replace resource %s",
+                  fedoraResource.getId()), ex);
+        }
+
+
+    }
+    /**
+     * Parse the request body as a Model.
+     *
+     * @param requestBodyStream rdf request body
+     * @param contentType content type of body
+     * @param resource the fedora resource
+     * @return Model containing triples from request body
+     * @throws MalformedRdfException in case rdf json cannot be parsed
+     */
+    private Model parseBodyAsModel(final InputStream requestBodyStream,
+            final String contentType, final FedoraResource resource) throws MalformedRdfException {
+        final Lang format = contentTypeToLang(contentType);
+
+        final Model inputModel;
+        try {
+            inputModel = createDefaultModel();
+            inputModel.read(requestBodyStream, "www.example.com", format.getName().toUpperCase());
```

Comment:
don't think `"www.example.com"` is what we want here, i think we want the base to be the uri of the resource. Although this might be a problem since the the RDF will probably have the HTTP URI for the resource and the kernel won't know how to convert that to an internal fedora uri... I wonder if model parsing needs to happen in the http layer because of this.

Reply:
I moved this same method down a layer so we'll need to merge our uses.

As for the other thing, I am of the mind that in the HTTP layer we parse the body and do URI replacement for any URI in the domain of the repository. Then pass it along so at this level we might have `info:fedora/some/object` as the URI.
ie.
```
 protected Model parseBodyAsModel(final InputStream requestBodyStream,
                                   final String contentType, final String fedoraId) throws MalformedRdfException {
```

Reply:
Reworked.